### PR TITLE
Social media website - Milestone: friendships v1

### PR DESCRIPTION
**Pull request by Abhigyan001**
In this milestone, we implemented the following features for a logged-in user :

A user is able to see all users list.
A user is able to send a friendship invitation.
A user can see a button “Send Friend Request” next to the name of the user who is not his friend yet.
A user is able to see pending friendship invitations sent to him from other users.
A user is able to accept or reject friendships invitation.
A user is able to see his friends.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
-ruby '2.7.0'
+ruby '>=2.6.6'
```
Comment (Dandush03) [inline Gemfile:4]: - [x] **Inconsisty ruby version:** your `Gemfile` and your `.ruby-version` have 2 different ruby version which course issues when running blunder, especially if the user has multiple environments

File: config/database.yml [hunk 2]
```diff
@@ -22,8 +22,14 @@ default: &default
...
-  <<: *default
-  database: ror_social_scaffold_development
+  <<: *default  
+  adapter: postgresql
+  username: postgres
+  password: 12345
+  host: localhost
+  port: 5432
+  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
+  timeout: 5000
```
Comment (Dandush03) [inline config/database.yml]: - [x] **Database Configuration:** I got multiple errors while trying to migrate your database due to the fact that you've set most of the pg configuration for development. let's make it work in the test environment and the production environment
```yml
default: &default
  adapter: postgresql
  username: postgres
  password: 12345
  encoding: unicode
  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
  host: localhost
  port: 5432

development:
  <<: *default
  database: ror_social_scaffold_development
```

**Review by Dandush03 — CHANGES_REQUESTED** [review]
# Status: Require Changes :stop_sign: 

### Great Work So Far :muscle: 
**Please fix all the issues pointed below before submitting for a new review**
- [x] **All User Tab:** Should not display current user
- [x] **Post Friend and Logged User:** Posts timeline includes posts written by logged-in user and their friends
- [x] **Tests:** Make sure you test each route/model/controller/view of the application
- [x] **Inconsisty ruby version:** check my comments [here](https://github.com/Abhigyan001/ror-social-scaffold/pull/2#discussion_r521295043)
- [x] **Database Configuration:** check my comments [here](https://github.com/Abhigyan001/ror-social-scaffold/pull/2#discussion_r521299183)

# Happy Coding :nerd_face: :computer:

**Review by elmejdki — CHANGES_REQUESTED** [review]
# status: require changes :recycle: 
## Hello guys :wave: :wave: :wave:, you still have some work :hammer: :toolbox: :wrench: to do, please make all the changes commented bellow.

- [ ] A  user should be able to reject a friend request:
image

- [ ] User should be able to see pending request:
image

- [ ] A user should be able to accept or reject a request:
image

- [ ] show an add friend button on the user page:
image

- [ ] As mentioned by the previous TSE:
image

- [ ] Add a linter to your project it's not configured yet.

### please make all the required changes before you submit another code review.
## Happy Coding :tophat:

**Review by rOluochKe — APPROVED** [review]
Hello Team
### Well done :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy coding!

**Pull request merged**